src/clashing_inputs.rs

```diff
@@ -452,55 +322,51 @@ mod tests {
     }
 
     mod basic_functionality {
-        use crate::axislike::VirtualDPad;
         use crate::buttonlike::ButtonState;
-        use crate::input_mocking::MockInput;
-        use bevy::input::InputPlugin;
+        use crate::input_like::chords::Chord;
+        use crate::input_like::virtual_dpad::VirtualDPad;
+        use bevy::input::{Input, InputPlugin};
+        use bevy::prelude::KeyCode;
         use Action::*;
 
         use super::*;
 
         #[test]
         fn clash_detection() {
-            let a: UserInput = A.into();
-            let b: UserInput = B.into();
-            let c: UserInput = C.into();
-            let ab = UserInput::chord([A, B]);
-            let bc = UserInput::chord([B, C]);
-            let abc = UserInput::chord([A, B, C]);
-            let axyz_dpad: UserInput = VirtualDPad {
+            let ab = Chord::new([A, B]);
+            let bc = Chord::new([B, C]);
+            let abc = Chord::new([A, B, C]);
+            let axyz_dpad = VirtualDPad {
                 up: A.into(),
                 down: X.into(),
                 left: Y.into(),
                 right: Z.into(),
-            }
-            .into();
-            let abcd_dpad: UserInput = VirtualDPad {
+            };
+            let abcd_dpad = VirtualDPad {
                 up: A.into(),
                 down: B.into(),
                 left: C.into(),
                 right: D.into(),
-            }
-            .into();
+            };
 
-            let ctrl_up: UserInput = UserInput::chord([Up, LControl]);
-            let directions_dpad: UserInput = VirtualDPad {
+            let ctrl_up = Chord::new([Up, LControl]);
+            let directions_dpad = VirtualDPad {
                 up: Up.into(),
                 down: Down.into(),
                 left: Left.into(),
                 right: Right.into(),
-            }
-            .into();
+            };
 
-            assert!(!a.clashes(&b));
-            assert!(a.clashes(&ab));
-            assert!(!c.clashes(&ab));
+            assert!(!A.clashes(&B));
+            assert!(A.clashes(&ab));
+            assert!(!C.clashes(&ab));
             assert!(!ab.clashes(&bc));
             assert!(ab.clashes(&abc));
-            assert!(axyz_dpad.clashes(&a));
+            // VirtualDPads are considered single inputs so they don't clash with other single inputs
+            assert!(!axyz_dpad.clashes(&A));
             assert!(axyz_dpad.clashes(&ab));
             assert!(!axyz_dpad.clashes(&bc));
-            assert!(axyz_dpad.clashes(&abcd_dpad));
+            assert!(!axyz_dpad.clashes(&abcd_dpad));
             assert!(ctrl_up.clashes(&directions_dpad));
```

Comment:
@alice-i-cecile I inverted two assertions here. It seemed to make more sense to me and was easier to implement if VirtualDPads don't clash with other single inputs since they are considered having a length of 1 and other single inputs don't clash with each other.

It seems like the concept of "clashes" is more accurately "conflicts" where it's purpose isn't to prevent the same input from being mapped to other things, but to detect when a key combination overlaps another input and prioritize longer combinations. For virtual dpads it seems like it would be more consistent to treat them the same as single inputs in regards to clashes. 

Let me know if there was a reason for this or if you don't agree and I can work in an exception to how they are handled.

Reply:
Yeah, that seems like a better approach.